src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/GetAzureRmMlOpClusterKey.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.MachineLearningCompute;
+using Microsoft.Azure.Management.MachineLearningCompute.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.MachineLearningCompute.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Get, CmdletSuffix + "Key")]
+    [OutputType(typeof(OperationalizationClusterCredentials))]
+    public class GetAzureRmMlOpClusterKey : MachineLearningComputeCmdletBase
+    {
+        protected const string CmdletParametersParameterSet =
+            "Get operationalization cluster's keys from cmdlet input parameters.";
+
+        protected const string ObjectParameterSet =
+            "Get operationalization cluster's keys from an OperationalizationCluster instance definition.";
+
+        protected const string ResourceIdParameterSet =
+            "Get operationalization cluster's keys from an Azure resource id.";
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = true,
+            HelpMessage = ResourceGroupParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = true, 
+            HelpMessage = NameParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ObjectParameterSet,
+            Mandatory = true, 
+            ValueFromPipeline = true,
+            HelpMessage = ClusterObjectParameterHelpMessage)]
+        [Alias(ClusterInputObjectAlias)]
+        public PSOperationalizationCluster InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true, 
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = ResourceIdParameterHelpMessage)]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (string.Equals(this.ParameterSetName, ObjectParameterSet, StringComparison.OrdinalIgnoreCase))
+            {
+                var resourceInfo = new ResourceIdentifier(InputObject.Id);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                Name = resourceInfo.ResourceName;
+            }
+            else if (string.Equals(this.ParameterSetName, ResourceIdParameterSet, StringComparison.OrdinalIgnoreCase))
+            {
+                var resourceInfo = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                Name = resourceInfo.ResourceName;
+            }
+
+            try
+            {
+                WriteObject(MachineLearningComputeManagementClient.OperationalizationClusters.ListKeys(ResourceGroupName, Name));
```

Comment:
@shutchings if this is returning more than one object, make sure to  use the `WriteObject` overload with `enumerateCollection:true`

Reply:
This method only ever returns one object that wraps up multiple keys.